### PR TITLE
Replaced `scala-mode2` with `ensime` package.

### PR DESCRIPTION
**Pull request by inflation**
* Since `scala-mode2` is not available on MELPA and is not activily developing.
* `ensime` package is accompanied with `scala-mode` and `sbt-mode`.
* Modified .gitignore file to ignore `target/` folder that is created by `sbt-mode`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,4 @@ session*
 .cask
 tramp
 /var/pcache
+target/
```
Comment (bbatsov) [inline .gitignore]: Why is this needed?

File: modules/prelude-scala.el [hunk 2]
```diff
@@ -33,10 +33,10 @@
 ;;; Code:
 
 (require 'prelude-programming)
-(prelude-require-packages '(scala-mode2))
+(prelude-require-packages '(ensime))
```
Comment (bbatsov) [inline modules/prelude-scala.el]: Btw, doesn't this package require some additional manual setup?

**Comment by inflation** [comment]
The `.gitignore` file does not need to modify, that's an accidentally activating `ensime` in the `.emacs.d` folder.

File: modules/prelude-scala.el (hunk 2 above)
Comment (inflation) [inline modules/prelude-scala.el]: The basic functions like syntax highlighting are automatically active when you install the package. And more complex functions do need manual init regarding which manage tool you are using.
